### PR TITLE
Fix url rewrite starting with numbers

### PR DESCRIPTION
**Pull request by antoin-m**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | URL Rewrites starting with numbers and dash were breaking the front for product without combinations.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-2393
| How to test?  | Create a product without combinations. Give it a price and a quantity. Set the url rewrite to `5-{product_name}` and go to its front page. You should be able to buy the product.

### CODE REVIEW

File: controllers/front/ProductController.php
Comment (antoin-m) [inline controllers/front/ProductController.php]: Without it the URL looks really dirty.

With:      http://localhost/PrestaShop/home/8-2-test-no-combi.html
Without: http://localhost/PrestaShop/home/8-2-2-test-no-combi.html?id_product_attribute=2

Maybe there is a better way/place to do it? @maximebiloe @aleeks

**Comment by vincentbz** [comment]
hmm not working :p

**Comment by antoin-m** [comment]
Ok so here is the situation:

A product's URL can take up to 4 forms matching this dispatcher rule : `{category:/}{id}{-:id_product_attribute}-{rewrite}`
Let's say `1` is the product's id, `2` is the id_product_attribute and `3` is the url rewrite.
You can then have :
`1-2` -> Coming on the product page from the front end with no URL rewrite
`1-2-3` -> Coming on the product page from the front end with a URL rewrite
`1` -> Coming on the product page from the preview button with no URL rewrite
`1-3` -> Coming on the product page from the preview button with a URL rewrite

The modification adds the id_product_attribute in the preview link so that URLs will be similar to the ones on the front. (top 2 ones)
Now the thing is... The preview link isn't refreshed when you save your product and thus it won't work if you transform your product from a simple product to a product with combinations. You just have to refresh the page but that doesn't seem to be an option. ^^"

So... Any ideas? @maximebiloe @aleeks

**Comment by aleeks** [comment]
Related: https://github.com/PrestaShop/PrestaShop/pull/7548

@antoin-m  take a break, take a kitkat..

**Comment by antoin-m** [comment]
UPDATE: I changed some stuff and took with me a commit from https://github.com/PrestaShop/PrestaShop/pull/5342 and I think it finally works! I sent it to @vincentbz for review.

If this PR is merged you can close https://github.com/PrestaShop/PrestaShop/pull/5342

**Review by maximebiloe — APPROVED** [review]

**Comment by maximebiloe** [comment]
Seems fine to me. Now waiting for @vincentbz approval 😉

**Comment by antoin-m** [comment]
image

**Comment by vincentbz** [comment]
giphybear

**Comment by maximebiloe** [comment]
Thank you @antoin-m

**Comment by Soullivaneuh** [comment]
Thank you @antoin-m for my commit integration.

I guess I can close #5342 now.

**Pull request merged**
